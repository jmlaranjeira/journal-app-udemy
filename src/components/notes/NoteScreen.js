

import React from 'react'
import { NotesAppBar } from "./NotesAppBar"

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
        
        <NotesAppBar />

        <div className="notes__content">

            <input type="text" placeholder="Some awesome title" className="notes__title-input" autoComplete="off"/>
            <textarea placeholder="What happened today" className="notes__textarea"></textarea>
            <div className="notes__image">
                <img src="https://www.tooltyp.com/wp-content/uploads/2014/10/1900x920-8-beneficios-de-usar-imagenes-en-nuestros-sitios-web.jpg" alt="imagen" />
            </div>
        </div>
    </div>
  )
}
