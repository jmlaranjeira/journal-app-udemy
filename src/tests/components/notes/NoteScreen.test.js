import React from 'react'
import { mount } from "enzyme"
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from 'redux-mock-store' //ES6 modules
import '@testing-library/jest-dom';
import { NoteScreen } from "../../../components/notes/NoteScreen";
import { activeNote } from "../../../actions/notes";


jest.mock("../../../actions/notes", () => ({
    activeNote: jest.fn()
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

let initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 'ABC',
            title: 'Hola',
            body: 'Mundo',
            date: 0
        },
        notes: []
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const userLogin = {
    email: 'bertin@gmail.com', 
    password: '123456'
}

const wrapper = mount(
    <Provider store={ store }>
        <NoteScreen />
    </Provider>
);
describe('Pruebas sobre el componente <NoteScreen />', () => {

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de disparar el activeNote', () => {

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        });

        expect( activeNote ).toHaveBeenLastCalledWith( initState.notes.active.id, {
            title: 'Hola de nuevo',
            body: initState.notes.active.body,
            date: initState.notes.active.date,
            id: initState.notes.active.id
        } )
    })
})