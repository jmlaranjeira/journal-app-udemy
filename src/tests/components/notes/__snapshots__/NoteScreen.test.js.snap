// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <NoteScreen /> Debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 de agosto 2020
    </span>
    <input
      id="file-selector"
      name="file-selector"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hola"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="Mundo"
    />
  </div>
  <button
    className="btn btn-danger btn-block"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</div>
`;
